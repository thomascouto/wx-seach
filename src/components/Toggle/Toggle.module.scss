.toggleButtons {
  button {
    color: #96a7be;
	height: 100%;
	padding: 9.5px 12px;
	border: 2px solid #eaedf4;
	display: inline-block;
	cursor: pointer;
	transition: all 0.3s ease-in-out;

	&:hover {
		filter: brightness(0.7);
	}
  }
}

.active {
  filter: brightness(0.7);
}