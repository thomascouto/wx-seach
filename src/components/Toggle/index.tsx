import clsx from 'clsx';

import styles from './Toggle.module.scss';

interface ToggleProps {
  handleMetricSystem(system: Units): void;
  active: Units;
}

export default function Toggle({ handleMetricSystem, active }: ToggleProps) {
  return (
    <div className={styles.toggleButtons}>
      <button
        className={clsx(active === 'metric' && styles.active)}
        onClick={() => {
          handleMetricSystem('metric');
        }}
      >
        °C
      </button>
      <button
        className={clsx(active === 'imperial' && styles.active)}
        onClick={() => {
          handleMetricSystem('imperial');
        }}
      >
        °F
      </button>
    </div>
  );
}
