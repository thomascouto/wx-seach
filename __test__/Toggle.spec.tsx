import { render, screen } from '@testing-library/react';

import Toggle from '@/components/Toggle';

describe('Toggle tests', () => {
  it('should imperial be active', () => {
    const { container } = render(<Toggle active="imperial" handleMetricSystem={jest.fn} />);

    const [metric, imperial] = screen.getAllByRole('button');

    expect(metric.classList.length).toEqual(0);
    expect(imperial.classList.item(0)).toEqual('active');

    expect(container).toMatchSnapshot();
  });

  it('should metric be active', () => {
    const { container } = render(<Toggle active="metric" handleMetricSystem={jest.fn} />);

    const [metric, imperial] = screen.getAllByRole('button');

    expect(metric.classList.item(0)).toEqual('active');
    expect(imperial.classList.length).toEqual(0);

    expect(container).toMatchSnapshot();
  });
});
