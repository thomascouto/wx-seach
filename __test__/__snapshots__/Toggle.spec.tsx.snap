// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle tests should imperial be active by default 1`] = `
<div>
  <div
    class="toggleButtons"
  >
    <button
      class=""
    >
      째C
    </button>
    <button
      class="active"
    >
      째F
    </button>
  </div>
</div>
`;

exports[`Toggle tests should metric be active by default 1`] = `
<div>
  <div
    class="toggleButtons"
  >
    <button
      class="active"
    >
      째C
    </button>
    <button
      class=""
    >
      째F
    </button>
  </div>
</div>
`;
